# Heart Disease

data source : https://archive.ics.uci.edu/dataset/45/heart+disease

Let's import the necessary libraries and load the data:

In [7]:
import pandas as pd

data = pd.read_csv('heart_disease_data.csv')

Here is an overview of the data:

In [8]:
data

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0    63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1    67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2    67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3    37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4    41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   
..    ...  ...  ...       ...    ...  ...      ...      ...    ...      ...   
292  57.0  0.0  4.0     140.0  241.0  0.0      0.0    123.0    1.0      0.2   
293  45.0  1.0  1.0     110.0  264.0  0.0      0.0    132.0    0.0      1.2   
294  68.0  1.0  4.0     144.0  193.0  1.0      0.0    141.0    0.0      3.4   
295  57.0  1.0  4.0     130.0  131.0  0.0      0.0    115.0    1.0      1.2   
296  57.0  0.0  2.0     130.0  236.0  0.0      2.0    174.0    0.0      0.0   

     slope   ca  thal  diagnosis  
0      3.0  0.0   6.0          0  
1      2.0  3.0   3.0          1  
2      2.0  2.0   7.0          1  
3      3.0  0.0   3.0          0  
4      1.0  0.0   3.0          0  
..     ...  ...   ...        ...  
292    2.0  0.0   7.0          1  
293    2.0  0.0   7.0          1  
294    2.0  2.0   7.0          1  
295    2.0  1.0   7.0          1  
296    2.0  1.0   3.0          1  

[297 rows x 14 columns]

We will use diagnosis as the variable we want to predict. There will be relationships between most of the variables listed above and whether or not the patient has got heart disease. We're going to explore the variables and attempt to replicate the distribution of the data in relation to diagnosis.

These are the descriptions of our attributes:
- **AGE:** in years
- **SEX:** 1 = male, 0 = female
- **CP:** Chest pain
  1. 1 = Typical angina
  2. 2 = Atypical angina
  3. 3 = Non-anginal pain
  4. 4 = Asymptomatic
- **TRESTBPS:** Resting blood pressure (in mg Hg on admission to the hospital)
- **CHOL:** serum cholesterol in mg/dL
- **FBS:** (fasting blood sugar > 120 mg/dL) (1 = true; 0 = false)
- **RESTECG:** resting electrocardiographic results
- **THALACH:** maximum heart rate achieved
- **EXANG:** exercise-induced angina (1 = yes; 0 = no)
- **OLDPEAK:** ST depression induced by exercise relative to rest
- **SLOPE:** the slope of the peak exercise ST segment
  1. upsloping
  2. flat
  3. downsloping
- **CA:** number of major vessels (0-3) colored by fluoroscopy
- **THAL:** 3 = normal; 6 = fixed defect; 7 = reversible defect
- **DIAGNOSIS (1 = yes; 0 = no):**